core/src/main/java/org/ehcache/core/Ehcache.java

```diff
@@ -1079,4 +1079,25 @@ public AtomicInteger getActualRemoveCount() {
     }
   }
 
+  @Override
+  public String toString() {
+    return "statusTransitioner:\n    "
+        + statusTransitioner.toString().replace("\n","\n    ") + "\n"
+        + "store:\n    "
+        + store.toString().replace("\n","\n    ") ;
+//        ", resilienceStrategy=" + resilienceStrategy +
+//        ", runtimeConfiguration=" + runtimeConfiguration +
+//        ", jsr107Cache=" + jsr107Cache +
+//        ", getObserver=" + getObserver +
+//        ", getAllObserver=" + getAllObserver +
+//        ", putObserver=" + putObserver +
+//        ", putAllObserver=" + putAllObserver +
+//        ", removeObserver=" + removeObserver +
```

Comment:
For observers and listeners, I wouldn't `toString()` them. But just knowing some have been setup is helpful for troubleshooting.  So perhaps something like `", getAllObserver=" + getAllObserver.size() +` and `cacheManagerListeners=` + cacheManagerListeners.size()` would be enough ?